Source/WebCore/platform/audio/gstreamer/AudioSourceProviderGStreamer.cpp

```diff
@@ -122,18 +123,23 @@ AudioSourceProviderGStreamer::AudioSourceProviderGStreamer(MediaStreamTrackPriva
     ASSERT(bus);
 
     gst_bus_set_sync_handler(bus.get(), [](GstBus*, GstMessage* message, gpointer userData) -> GstBusSyncReply {
-        auto* decodebin = GST_ELEMENT_CAST(userData);
-        if (GST_MESSAGE_TYPE(message) != GST_MESSAGE_STREAM_COLLECTION || GST_MESSAGE_SRC(message) != GST_OBJECT_CAST(decodebin)) {
+        auto scopeExit = makeScopeExit([&] {
```

Comment:
I think the proper way of doing this would be adopting the message into a GRefPtr.